lib/google/gapic/schema/wrappers.rb

```diff
@@ -0,0 +1,225 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'protobuf/descriptors'
+
+module Google
+  module Gapic
+    module Schema
+      # Base class for all generic proto types including: enums, messages, methods
+      # and services.
+      #
+      # @!attribute [r] descriptor
+      #   @ return [Object] the descriptor of the proto
+      # @!attribute [r] address
+      #   @ return [Enumerable<String>] The address of the proto.
+      class Proto
+        extend Forwardable
+        attr_reader :descriptor, :address, :docs
+
+        # Initializes a Proto object.
+        # @param descriptor [Object] the protobuf
+        #   representation of this service.
+        def initialize(descriptor, address, docs)
+          @descriptor = descriptor
+          @address = address
+          @docs = docs
+        end
+
+        def_delegators(
+          :docs,
+          *Google::Protobuf::SourceCodeInfo::Location.all_fields.map{ |f| f.name.to_sym })
+      end
+
+      # Wrapper for aprotobuf service.
+      #
+      # @!attribute [r] methods
+      #   @ return [Enumerable<Method>] the methods of this service.
+      class Service < Proto
+        extend Forwardable
+        attr_reader :methods
+
+        # Initializes a Service object.
+        # @param descriptor [Google::Protobuf::ServiceDescriptorProto] the protobuf
+        #   representation of this service.
+        def initialize(descriptor, address, docs, methods)
+          super(descriptor, address, docs)
+          @methods = methods || {}
+        end
+
+        # Delegate the getters for the descriptor fields to the descriptor.
+        def_delegators(
```

Comment:
Let's enumerate these like we discussed since the underlying `Google::Protobuf::ServiceDescriptorProto` isn't documented very well (I'd suggested doing this as a follow on PR though).

Reply:
I updated the docs just now.